File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Pages/getpageobject.rb
module Asposediagramjava
  module GetPageObject
    def initialize()
        # Getting a Page Object by ID
        get_page_object_by_id()

        # Getting a Page Object by Name
        get_page_object_by_name()        
    end

    def get_page_object_by_id() 
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        page_id = 0
        # Get page object by id
        page = diagram.getPages().getPage(page_id)

        puts "Page : " + page.to_string
    end

    def get_page_object_by_name() 
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Set page name
        page_name = "Flow 1"
        # Get page object by name
        page = diagram.getPages().getPage(page_name)

        puts "Page : " + page.to_string
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithPages/GetPageObject.php
<?php
namespace Aspose\Diagram\WorkingwithPages;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class GetPageObject{

    public static function run($dataDir=null){

        # Getting a Page Object by ID
        GetPageObject::get_page_object_by_id($dataDir);

        # Getting a Page Object by Name
        GetPageObject::get_page_object_by_name($dataDir);
    }

    public static function get_page_object_by_id($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        $page_id = 0;
        # Get page object by id
        $page = $diagram->getPages()->getPage($page_id);
        print "Page : ".(string)$page.PHP_EOL;
    }

    public static function get_page_object_by_name($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        # Set page name
        $page_name = "Flow 1";
        # Get page object by name
        $page = $diagram->getPages()->getPage($page_name);
        print "Page : ".(string)$page.PHP_EOL;
    }
}